src/cbc_sdk/endpoint_standard/usb_device_control.py

```diff
@@ -0,0 +1,815 @@
+#!/usr/bin/env python3
+
+# *******************************************************
+# Copyright (c) VMware, Inc. 2020. All Rights Reserved.
+# SPDX-License-Identifier: MIT
+# *******************************************************
+# *
+# * DISCLAIMER. THIS PROGRAM IS PROVIDED TO YOU "AS IS" WITHOUT
+# * WARRANTIES OR CONDITIONS OF ANY KIND, WHETHER ORAL OR WRITTEN,
+# * EXPRESS OR IMPLIED. THE AUTHOR SPECIFICALLY DISCLAIMS ANY IMPLIED
+# * WARRANTIES OR CONDITIONS OF MERCHANTABILITY, SATISFACTORY QUALITY,
+# * NON-INFRINGEMENT AND FITNESS FOR A PARTICULAR PURPOSE.
+
+"""Model and Query Classes for USB Device Control"""
+
+from cbc_sdk.base import (NewBaseModel, MutableBaseModel, BaseQuery, QueryBuilder, QueryBuilderSupportMixin,
+                          IterableQueryMixin, AsyncQueryMixin)
+from cbc_sdk.errors import ApiError, ServerError
+from cbc_sdk.platform.devices import DeviceSearchQuery
+import logging
+import time
+import json
+
+log = logging.getLogger(__name__)
+
+"""USB Device Control models"""
+
+
+class USBDeviceApproval(MutableBaseModel):
+    """Represents a USB device approval."""
+    urlobject = "/device_control/v3/orgs/{0}/approvals"
+    urlobject_single = "/device_control/v3/orgs/{0}/approvals/{1}"
+    primary_key = "id"
+    swagger_meta_file = "endpoint_standard/models/usbDeviceApproval.yaml"
+
+    def __init__(self, cb, model_unique_id, initial_data=None):
+        """
+        Initialize the USBDeviceApproval object.
+
+        Args:
+            cb (BaseAPI): Reference to API object used to communicate with the server.
+            model_unique_id (str): ID of the alert represented.
+            initial_data (dict): Initial data used to populate the alert.
+        """
+        super(USBDeviceApproval, self).__init__(cb, model_unique_id, initial_data)
+        if model_unique_id is not None and initial_data is None:
+            self._refresh()
+
+    @classmethod
+    def _query_implementation(cls, cb, **kwargs):
+        """
+        Returns the appropriate query object for this object type.
+
+        Args:
+            cb (BaseAPI): Reference to API object used to communicate with the server.
+            **kwargs (dict): Not used, retained for compatibility.
+
+        Returns:
+            USBDeviceApprovalQuery: The query object for this alert type.
+        """
+        return USBDeviceApprovalQuery(cls, cb)
+
+    def _build_api_request_uri(self, http_method="GET"):
+        """
+        Build the unique URL used to make requests for this object.
+
+        Args:
+            http_method (str): Not used; retained for compatibility.
+
+        Returns:
+            str: The URL used to make requests for this object.
+        """
+        return self.urlobject_single.format(self._cb.credentials.org_key, self._model_unique_id)
+
+    def _refresh(self):
+        """
+        Rereads the object data from the server.
```

Comment:
This is a mutable model so you will need to set full_init to true as this relates to the __str__ message that is appended when not true. See the section I modified for reference https://github.com/carbonblack/carbon-black-cloud-sdk-python/pull/104

Reply:
The same will apply to any Block as well

Reply:
This is the only mutable model; is it necessary to set _full_init in any non-mutable models?

Reply:
If you inherit the UnrefreshableModel then No since I added the additional condition to the str function